spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonClientConfiguration.java

```diff
@@ -200,7 +200,11 @@ protected OverrideRestClient(IClientConfig ncc,
 		public URI reconstructURIWithServer(Server server, URI original) {
 			String scheme = original.getScheme();
 			if (!"https".equals(scheme) && this.serverIntrospector.isSecure(server)) {
-				original = UriComponentsBuilder.fromUri(original).scheme("https").build(true)
+				original = UriComponentsBuilder.fromUri(original)
+						.scheme("https")
+						.host(server.getHost())
+						.port(server.getPort())
+						.build(true)
 						.toUri();
 			}
 			return super.reconstructURIWithServer(server, original);
```

Comment:
I ask because if host and port of the `server` match original, none of the logic in super is executed, which is what this change does.